feat: rename refresh to rebuild in `ddev debug refresh`, allow to build different services, add cache flag

stasadev opened the pull request:
## The Issue

`ddev debug refresh` is not flexible enough. It runs rebuild without cache only for the web service.

## How This PR Solves The Issue

- adds `--service=name` and `--all` flag to build specified service.
- adds `--cache` flag that will not add `--no-cache` to the underlying `docker-compose build` command.
- adds `--progress plain` to `docker-compose build`, because it was squashed by default.

## Manual Testing Instructions

```
ddev debug refresh # this command is the same as it was before
ddev debug refresh --cache # should show cached output
ddev debug refresh --all # should rebuild all services (if you have web, db, redis: all of them should be rebuilt)
ddev debug refresh --service db # the same as usual command, but for the db service
```

## Automated Testing Overview

<!-- Please describe the tests introduced by this PR, or explain why no tests are needed. -->

## Release/Deployment Notes

<!-- Does this affect anything else or have ramifications for other code? Does anything have to be done on deployment? -->

rfay commented:
Wow, exciting!

rfay commented on cmd/ddev/cmd/debug-refresh.go:
```diff
@@ -12,12 +13,17 @@ import (
 )
 
+var (
+	buildAll bool
```
In general, I don't like these globals for this, and prefer checking the flags as we do in most places. Of course, this is just in `ddev debug` so not important at all, so no big deal.

stasadev replied:
The variable here is only needed because the flag has a short and long form, I just copied the logic from some other command.

rfay commented on cmd/ddev/cmd/debug-refresh.go:
```diff
@@ -12,12 +13,17 @@ import (
+
 // DebugRefreshCmd implements the ddev debug refresh command
 var DebugRefreshCmd = &cobra.Command{
 	ValidArgsFunction: ddevapp.GetProjectNamesFunc("all", 1),
```
`refresh` was a great name for this when it was created, but how about changing it to `rebuild` with `refresh` being an alias?

stasadev replied:
Okay, `rebuild` gives more context, I'll change it.

rfay reviewed (approved):
Super nice, can't wait to have this!

stasadev commented:
> `refresh` was a great name for this when it was created, but how about changing it to `rebuild` with `refresh` being an alias?

Renamed, added the alias, and updated the documentation.
